src/Utilities.UnitTests/Options/SymbolNamesWithValueOptionTests.cs

```diff
@@ -19,7 +19,7 @@ public sealed class SymbolNamesWithValueOptionTests
         public void WhenNoSymbolNames_ReturnsEmpty()
         {
             // Arrange & act
-            var options = SymbolNamesWithValueOption<Unit>.Create(ImmutableArray<string>.Empty, GetCompilation(), null, null);
+            var options = SymbolNamesWithValueOption<Unit>.Create(ImmutableArray<string>.Empty, GetCompilation(), null, static name => new SymbolNamesWithValueOption<Unit>.NameParts(name, Unit.Default));
```

Comment:
That looks like a possible good improvement for the make static rule, WDYT?

Reply:
I like it, but maybe should be a separate diagnostic ID so it can be turned off?

Reply:
I am actually curious if it would be hard technically to allow fine-tuning of rule descriptor through the `.editorconfig` so that we can keep rules with a same scope under the same ID but allow specific descriptors to be tweaked.

For example, `dotnet_code_quality.CAXXXX.severity = warning` would keep the same behavior (it controls all descriptor for this ID) but `dotnet_code_quality.CAXXXX-desc1.severity = warning` only change the severity of one of the descriptors.

Reply:
Or something like https://github.com/dotnet/roslyn/issues/27712?

Reply:
I believe @sharwell has a compiler design proposal for allowing sub-diagnostics for similar configuration - different severity buckets for same ID. We should wait for that feature before attempting anything similar ourselves from an analyzer package.

Reply:
Of course you thought about it before me xD And I even upvoted but forgot about it.